simulation/traffic_simulator/include/traffic_simulator/entity/entity_base.hpp

```diff
@@ -113,7 +114,9 @@ class EntityBase
 
   virtual auto getObstacle() -> std::optional<traffic_simulator_msgs::msg::Obstacle> = 0;
 
-  virtual auto getRouteLanelets(const double horizon = 100) -> std::vector<std::int64_t> = 0;
+  virtual auto getRouteLanelets(double horizon = 100) const -> std::vector<std::int64_t> = 0;
+
+  virtual auto fillLaneletPose(traffic_simulator_msgs::msg::EntityStatus & status) const -> void;
```

Comment:
fill LaneletPose function should be pure virtual function because how to fill lanelet pose should be decided by entity type.